packages/components/calendar/src/date-table.ts

```diff
@@ -1,6 +1,40 @@
 import { buildProps, definePropType, isObject } from '@element-plus/utils'
+import { rangeArr } from '@element-plus/components/time-picker'
 import type { ExtractPropTypes } from 'vue'
 import type { Dayjs } from 'dayjs'
+import type DateTable from './date-table.vue'
+
+export type CellType = 'next' | 'prev' | 'current'
+export type Cell = {
+  text: number
+  type: CellType
+}
+
+export const WEEK_DAYS = [
```

Comment:
We can move it to `@element-plus/constants/date-pick.ts` (and maybe rename `date-pick.ts` to `date.ts`)